dask/dataframe/tests/test_arithmetics_reduction.py

```diff
@@ -1266,8 +1261,8 @@ def test_reductions_frame_dtypes_numeric_only():
         assert_eq(
             getattr(df, func)(**kwargs),
             getattr(ddf, func)(**kwargs),
-            check_dtypes=func in ["mean", "max"]
-            and (PANDAS_GT_120 or not PANDAS_GT_100),
+            check_dtypes=func in ["mean", "max"] and PANDAS_GT_120,
+            check_dtype=func in ["mean", "max"] and PANDAS_GT_120,
```

Comment:
Looks like the `check_dtype=...` line was added by mistake, no? It seems like `dask.dataframe.utils.assert_eq` only takes a `check_dtypes` (plural) kwarg.

Reply:
No it is forwarded by [`assert_eq`]( https://github.com/dask/dask/blob/a87236041ff223363698f65c5b7c153415a41259/dask/dataframe/utils.py#L559 ) to [`assert_series_equal`]( https://github.com/pandas-dev/pandas/blob/18d46b172e7eafec2c38780686ed2c71de7686f8/pandas/_testing/asserters.py#L857 ). Was adding this to try and fix a remaining test failure

Reply:
Is there a better way to forward that? It looks really strange to have two almost identical lines here, if I was looking at this code at a random point in time I was certainly think this is mistake.

Reply:
Not really no. Agree it is strange and don't like it much either, but we were already [doing that before]( https://github.com/dask/dask/blob/8f07a99f1c38f589804300bd3e8c3bf8dee7e657/dask/dataframe/tests/test_arithmetics_reduction.py#L1298 ) in the other line with this pattern below.

For now would rather focus on getting NumPy & Pandas updated. If someone wants to improve `assert_eq`, that would be welcome, but would prefer not to dig into that in this PR

Reply:
I agree, thanks for the details John!

Reply:
Filed as issue ( https://github.com/dask/dask/issues/7947 )